deep_sleep not working with esp32c3

### The problem

Been having issue with running deep_sleep, was hoping that https://github.com/esphome/esphome/pull/3066 would resolve the issue, but still having a problem after pulling latest dev branch with pip. Not sure if it's a local dev env issue or something else.




### Which version of ESPHome has the issue?

2022.3.0-dev

### What type of installation are you using?

pip

### Which version of Home Assistant has the issue?

N/A

### What platform are you using?

ESP32-IDF

### Board

esp32-c3-devkitm-1

### Component causing the issue

deep_sleep

### Example YAML snippet

```yaml
esphome:
  name: deep-sleep
esp32:
  board: esp32-c3-devkitm-1
  variant: esp32c3
  framework:
    type: esp-idf
deep_sleep:
  run_duration: 10s 
  sleep_duration: 20min
```


### Anything in the logs that might be useful for us?

```txt
src/esphome/components/deep_sleep/deep_sleep_component.cpp:78:6: error: no declaration matches 'void esphome::deep_sleep::DeepSleepComponent::set_ext1_wakeup(esphome::deep_sleep::Ext1Wakeup)'
 void DeepSleepComponent::set_ext1_wakeup(Ext1Wakeup ext1_wakeup) { this->ext1_wakeup_ = ext1_wakeup; }
      ^~~~~~~~~~~~~~~~~~
src/esphome/components/deep_sleep/deep_sleep_component.cpp:78:6: note: no functions named 'void esphome::deep_sleep::DeepSleepComponent::set_ext1_wakeup(esphome::deep_sleep::Ext1Wakeup)'
In file included from src/esphome/components/deep_sleep/deep_sleep_component.cpp:1:
src/esphome/components/deep_sleep/deep_sleep_component.h:56:7: note: 'class esphome::deep_sleep::DeepSleepComponent' defined here
 class DeepSleepComponent : public Component {
       ^~~~~~~~~~~~~~~~~~
src/esphome/components/deep_sleep/deep_sleep_component.cpp:79:6: error: no declaration matches 'void esphome::deep_sleep::DeepSleepComponent::set_touch_wakeup(bool)'
 void DeepSleepComponent::set_touch_wakeup(bool touch_wakeup) { this->touch_wakeup_ = touch_wakeup; }
      ^~~~~~~~~~~~~~~~~~
src/esphome/components/deep_sleep/deep_sleep_component.cpp:79:6: note: no functions named 'void esphome::deep_sleep::DeepSleepComponent::set_touch_wakeup(bool)'
In file included from src/esphome/components/deep_sleep/deep_sleep_component.cpp:1:
src/esphome/components/deep_sleep/deep_sleep_component.h:56:7: note: 'class esphome::deep_sleep::DeepSleepComponent' defined here
 class DeepSleepComponent : public Component {
       ^~~~~~~~~~~~~~~~~~
src/esphome/components/deep_sleep/deep_sleep_component.cpp:80:6: error: no declaration matches 'void esphome::deep_sleep::DeepSleepComponent::set_run_duration(esphome::deep_sleep::WakeupCauseToRunDuration)'
 void DeepSleepComponent::set_run_duration(WakeupCauseToRunDuration wakeup_cause_to_run_duration) {
      ^~~~~~~~~~~~~~~~~~
In file included from src/esphome/components/deep_sleep/deep_sleep_component.cpp:1:
src/esphome/components/deep_sleep/deep_sleep_component.h:81:8: note: candidate is: 'void esphome::deep_sleep::DeepSleepComponent::set_run_duration(uint32_t)'
   void set_run_duration(uint32_t time_ms);
        ^~~~~~~~~~~~~~~~
src/esphome/components/deep_sleep/deep_sleep_component.h:56:7: note: 'class esphome::deep_sleep::DeepSleepComponent' defined here
 class DeepSleepComponent : public Component {
       ^~~~~~~~~~~~~~~~~~
src/esphome/components/deep_sleep/deep_sleep_component.cpp: In member function 'void esphome::deep_sleep::DeepSleepComponent::begin_sleep(bool)':
src/esphome/components/deep_sleep/deep_sleep_component.cpp:137:81: error: cannot convert 'bool' to 'esp_deepsleep_gpio_wake_up_mode_t'
     esp_deep_sleep_enable_gpio_wakeup(gpio_num_t(this->wakeup_pin_->get_pin()), level);
                                                                                 ^~~~~
In file included from src/esphome/components/deep_sleep/deep_sleep_component.h:9,
                 from src/esphome/components/deep_sleep/deep_sleep_component.cpp:1:
C:/Users/Loki-VR/.platformio/packages/framework-espidf/components/esp_system/include/esp_sleep.h:262:103: note:   initializing argument 2 of 'esp_err_t esp_deep_sleep_enable_gpio_wakeup(uint64_t, esp_deepsleep_gpio_wake_up_mode_t)'
 esp_err_t esp_deep_sleep_enable_gpio_wakeup(uint64_t gpio_pin_mask, esp_deepsleep_gpio_wake_up_mode_t mode);
                                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~
*** [.pioenvs\deep-sleep\src\esphome\components\deep_sleep\deep_sleep_component.o] Error 1
=================================================================================== [FAILED] Took 15.25 seconds ===================================================================================
```


### Additional information

_No response_

--- a/esphome/components/deep_sleep/deep_sleep_component.cpp
+++ b/esphome/components/deep_sleep/deep_sleep_component.cpp
@@ -109,7 +109,7 @@ void DeepSleepComponent::begin_sleep(bool manual) {
 
   App.run_safe_shutdown_hooks();
 
-#if defined(USE_ESP32) 
+#if defined(USE_ESP32)
 #if !defined(USE_ESP32_VARIANT_ESP32C3)
   if (this->sleep_duration_.has_value())
     esp_sleep_enable_timer_wakeup(*this->sleep_duration_);
@@ -137,13 +137,12 @@ void DeepSleepComponent::begin_sleep(bool manual) {
     if (this->wakeup_pin_mode_ == WAKEUP_PIN_MODE_INVERT_WAKEUP && this->wakeup_pin_->digital_read()) {
       level = !level;
     }
-    esp_deep_sleep_enable_gpio_wakeup(gpio_num_t(this->wakeup_pin_->get_pin()), static_cast<esp_deepsleep_gpio_wake_up_mode_t>(level));
+    esp_deep_sleep_enable_gpio_wakeup(gpio_num_t(this->wakeup_pin_->get_pin()), 
+                                      static_cast<esp_deepsleep_gpio_wake_up_mode_t>(level));
   }
 #endif
   esp_deep_sleep_start();
 #endif
-
-
 
 #ifdef USE_ESP8266
   ESP.deepSleep(*this->sleep_duration_);  // NOLINT(readability-static-accessed-through-instance)

--- a/esphome/components/deep_sleep/deep_sleep_component.cpp
+++ b/esphome/components/deep_sleep/deep_sleep_component.cpp
@@ -76,11 +76,13 @@ void DeepSleepComponent::set_sleep_duration(uint32_t time_ms) { this->sleep_dura
 void DeepSleepComponent::set_wakeup_pin_mode(WakeupPinMode wakeup_pin_mode) {
   this->wakeup_pin_mode_ = wakeup_pin_mode;
 }
+#if !defined(USE_ESP32_VARIANT_ESP32C3)
 void DeepSleepComponent::set_ext1_wakeup(Ext1Wakeup ext1_wakeup) { this->ext1_wakeup_ = ext1_wakeup; }
 void DeepSleepComponent::set_touch_wakeup(bool touch_wakeup) { this->touch_wakeup_ = touch_wakeup; }
 void DeepSleepComponent::set_run_duration(WakeupCauseToRunDuration wakeup_cause_to_run_duration) {
   wakeup_cause_to_run_duration_ = wakeup_cause_to_run_duration;
 }
+#endif
 #endif
 void DeepSleepComponent::set_run_duration(uint32_t time_ms) { this->run_duration_ = time_ms; }
 void DeepSleepComponent::begin_sleep(bool manual) {
@@ -137,7 +139,7 @@ void DeepSleepComponent::begin_sleep(bool manual) {
     if (this->wakeup_pin_mode_ == WAKEUP_PIN_MODE_INVERT_WAKEUP && this->wakeup_pin_->digital_read()) {
       level = !level;
     }
-    esp_deep_sleep_enable_gpio_wakeup(gpio_num_t(this->wakeup_pin_->get_pin()), level);
+    esp_deep_sleep_enable_gpio_wakeup(gpio_num_t(this->wakeup_pin_->get_pin()), static_cast<esp_deepsleep_gpio_wake_up_mode_t>(level));
   }
 #endif
 

--- a/esphome/components/deep_sleep/deep_sleep_component.cpp
+++ b/esphome/components/deep_sleep/deep_sleep_component.cpp
@@ -137,7 +137,7 @@ void DeepSleepComponent::begin_sleep(bool manual) {
     if (this->wakeup_pin_mode_ == WAKEUP_PIN_MODE_INVERT_WAKEUP && this->wakeup_pin_->digital_read()) {
       level = !level;
     }
-    esp_deep_sleep_enable_gpio_wakeup(gpio_num_t(this->wakeup_pin_->get_pin()), 
+    esp_deep_sleep_enable_gpio_wakeup(gpio_num_t(this->wakeup_pin_->get_pin()),
                                       static_cast<esp_deepsleep_gpio_wake_up_mode_t>(level));
   }
 #endif

--- a/esphome/components/deep_sleep/deep_sleep_component.cpp
+++ b/esphome/components/deep_sleep/deep_sleep_component.cpp
@@ -109,7 +109,8 @@ void DeepSleepComponent::begin_sleep(bool manual) {
 
   App.run_safe_shutdown_hooks();
 
-#if defined(USE_ESP32) && !defined(USE_ESP32_VARIANT_ESP32C3)
+#if defined(USE_ESP32) 
+#if !defined(USE_ESP32_VARIANT_ESP32C3)
   if (this->sleep_duration_.has_value())
     esp_sleep_enable_timer_wakeup(*this->sleep_duration_);
   if (this->wakeup_pin_ != nullptr) {
@@ -127,10 +128,7 @@ void DeepSleepComponent::begin_sleep(bool manual) {
     esp_sleep_enable_touchpad_wakeup();
     esp_sleep_pd_config(ESP_PD_DOMAIN_RTC_PERIPH, ESP_PD_OPTION_ON);
   }
-
-  esp_deep_sleep_start();
 #endif
-
 #ifdef USE_ESP32_VARIANT_ESP32C3
   if (this->sleep_duration_.has_value())
     esp_sleep_enable_timer_wakeup(*this->sleep_duration_);
@@ -142,6 +140,10 @@ void DeepSleepComponent::begin_sleep(bool manual) {
     esp_deep_sleep_enable_gpio_wakeup(gpio_num_t(this->wakeup_pin_->get_pin()), static_cast<esp_deepsleep_gpio_wake_up_mode_t>(level));
   }
 #endif
+  esp_deep_sleep_start();
+#endif
+
+
 
 #ifdef USE_ESP8266
   ESP.deepSleep(*this->sleep_duration_);  // NOLINT(readability-static-accessed-through-instance)

